cas-server-core/src/main/java/org/jasig/cas/authentication/SuccessfulHandlerMetaDataPopulator.java

```diff
@@ -36,7 +37,12 @@
 
     @Override
     public void populateAttributes(final AuthenticationBuilder builder, final Credential credential) {
-        builder.addAttribute(SUCCESSFUL_AUTHENTICATION_HANDLERS, new HashSet(builder.getSuccesses().keySet()));
+        Set<String> successes = builder.getSuccesses().keySet();
```

Comment:
No it never should. When the populator runs, it is assumed that authN is successful. Need to double check, but I think that is the current behavior.